### PR TITLE
Fix `optimizer_hooks.GradientHardClipping` for scalar array

### LINKED ISSUE #7759: Refactor utility link in optimizer_hooks unit tests
`chainer_tests/optimizer_hooks_tests` uses similar dummy links, which can be refactored to reduce repetition.

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #7759

This PR was originally intended to refactor the tests (#7759) but I found a bug in `optimizer_hooks.GradientHardClipping`. I can split the PR if it's preferable.

### CODE REVIEW

**Comment by emcastillo** [comment]
The bug-fix is pretty small and related to the tests.
I think is ok to merge both of them together.

File: tests/chainer_tests/optimizer_hooks_tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import numpy
```
Comment (emcastillo) [inline tests/chainer_tests/optimizer_hooks_tests/utils.py]: Shouldnt this file be in `chainer/testing/` instead of tests

Comment (niboshi) [reply tests/chainer_tests/optimizer_hooks_tests/utils.py]: Hmm, I don't this this feature deserve a public API.

**Comment by niboshi** [comment]
Fixed errors.

**Comment by emcastillo** [comment]
Jenkins, test this please.

**Comment by emcastillo** [comment]
Jenkins, test this please.

**Comment by emcastillo** [comment]
Jenkins, test this please.

**Comment by emcastillo** [comment]
CI unrelated

**Comment by niboshi** [comment]
@emcastillo 
This is a bugfix. Can you backport?

**Comment by niboshi** [comment]
@emcastillo Can you make a backport?

**Pull request merged**